### ISSUE TITLE
[Bug]: incorrect duplicate route detection

### ISSUE THREAD

**Issue by rus-geek** [hashicorp/terraform-provider-aws#36510]
### Terraform Core Version

1.3.3

### AWS Provider Version

5.41.0

### Affected Resource(s)

aws_route

### Expected Behavior

Given a route table containing a propagated route to 0.0.0.0/0, I should be able to create a static route to 0.0.0.0/0 to override the propagated route.

### Actual Behavior

terraform refuses to create the route, incorrectly indicating that it is a duplicate

### Relevant Error/Panic Output Snippet

```shell
Error: RouteAlreadyExists: Route in Route Table (rtb-xxx) with destination (0.0.0.0/0) already exists
```

### Terraform Configuration Files

```terraform
resource "aws_route" "default_route" {
  route_table_id         = <your route table here>
  destination_cidr_block = "0.0.0.0/0"
  transit_gateway_id     = <your TGW here>
}
```

### Steps to Reproduce

Create a VPC with a propagated route to 0.0.0.0/0 with any next hop.
Attempt to add a static route to 0.0.0.0/0 with any next hop.

### Panic Output

Error: RouteAlreadyExists: Route in Route Table (rtb-xxx) with destination (0.0.0.0/0) already exists

### References

from the AWS documentation: "If the destination of a propagated route is identical to the destination of a static route, the static route takes priority..."

https://docs.aws.amazon.com/vpc/latest/userguide/VPC_Route_Tables.html#route-tables-priority

### Would you like to implement a fix?

None

**Comment by justinretzolk** [comment]
Hey @rus-geek 👋 Thank you for taking the time to raise this! I can't quite tell from the log snippet provided, but this error looks to be coming from the AWS API itself. Are you able to provide debug logging (redacted as needed) and/or test from the AWS CLI to see if it behaves the same? That's often a decent indicator in situations like this.

**Comment by rus-geek** [comment]
This works with prior versions of the provider, so it's allowed by AWS. The bug appears to have been introduced in this PR which "fixed" the creation of duplicate resources. https://github.com/hashicorp/terraform-provider-aws/pull/36326

**Comment by rus-geek** [comment]
The error message is being produced here: https://github.com/hashicorp/terraform-provider-aws/pull/36326/files#diff-75640a85d46fcb8ff09efe164520cac42e5e8ccb3c8d641bb96ef678ca472388R199

**Comment by ewbankkit** [comment]
@rus-geek Thanks for raising this issue 👏.
I can reproduce this for the very similar case of the VPC-local route added via `CreateRouteTable`.
The fix is to check that the duplicate route was created via `CreateRoute` and only fail then.
I can get this fixed for today's Terraform AWS Provider **v5.42.0** release.

**Comment by rus-geek** [comment]
Confirmed that this works as expected in 5.42.0. We are again able to create static routes that 'override' propagated routes.
